ignore/ignore.go

```diff
@@ -0,0 +1,73 @@
+// package ignore provides helpers to support ignore-files similar to .gitignore
+package ignore
+
+import (
+	"bufio"
+	"io"
+	"strings"
+)
+
+var (
+	LineComment = "#"
+	IgnoreFile  = ".sourcegraph/sourcegraphignore"
+)
+
+type IgnoreMatcher struct {
+	IgnoreList []string
```

Comment:
do we need to export this? Looks like users of this API only ever call `IgnoreMatcher.Match`

Reply:
The constructor of `ignoreMatcher` is not part of the ignore package and is expected to set the field. The constructor is specific to the indexer. For example, for searcher I will have to write another constructor.

Reply:
Edit: once we remove `strip`, we can return `Matcher` from `parseIgnoreFile`, i.e. the constructor doesn't need to access IgnoreList, i.e. we don't need to export it anymore :-)